### PR TITLE
Add ConsumerConfig and stream_with_config for using SmartStreams

### PR DESCRIPTION
**Pull request by nicholastmosher**
Closes https://github.com/infinyon/fluvio-client-wasm/issues/32

- Adds wrapper for `ConsumerConfig`
- Adds `streamWithConfig` for consumer, allowing passing WASM SmartStream via ConsumerConfig

### CODE REVIEW

File: src/consumer.rs [hunk 1]
```diff
@@ -52,7 +85,30 @@ impl PartitionConsumer {
             ))),
         })
     }
+
+    #[wasm_bindgen(js_name = "streamWithConfig")]
+    pub async fn stream_with_config(
```
Comment (simlay) [inline src/consumer.rs:90]: woah, this works?

Comment (simlay) [reply src/consumer.rs:90]: Wait, nvm, this takes ownership of `self`. I thought it was `&self`.

**Review by simlay — APPROVED** [review]
Other than broken CI, this looks good.

File: src/consumer.rs (hunk 1 above)
Comment (nicholastmosher) [reply src/consumer.rs:90]: Yeah 😂 why?

**Comment by nicholastmosher** [comment]
I was able to write this in a way that does not require changes to `fluvio`. Ready to merge if you're still good with it.

**Pull request merged**